imitation/reward_net.py

```diff
@@ -224,16 +247,50 @@ def __init__(self, env, *, state_only=False, theta_units=None, **kwargs):
     super().__init__(env, **kwargs)
 
   def build_theta_network(self, obs_input, act_input):
-    return _build_theta_network(self.state_only, self.theta_units,
-                                obs_input, self.env.observation_space,
-                                act_input, self.env.action_space)
+    act_or_none = None if self.state_only else act_input
+    return build_basic_theta_network(self.theta_units,
+                                     self.env.observation_space,
+                                     self.env.action_space,
+                                     old_obs_input=obs_input,
+                                     act_input=act_or_none,
+                                     new_obs_input=None)
 
   @property
   def reward_output_train(self):
     """Training reward is the same as the test reward, since no shaping."""
     return self.reward_output_test
 
 
+def build_basic_phi_network(hid_sizes, obs_space, old_obs_input, new_obs_input):
+  """Builds a potential network depending on specified observation.
+
+  Arguments:
+    hid_sizes (Optional[List[int]]): Number of units at each hidden layer.
+    obs_space (gym.Space): Observation space.
+    old_obs_input (Optional[tf.Tensor]): Previous observation.
+    new_obs_input (Optional[tf.Tensor]): Next observation.
+
+  Returns:
+    Tuple[tf.Tensor]: potential for the old and new observations.
```

Comment:
For documentation purposes this is fine, but fun-fact:
`Tuple[tf.Tensor, tf.Tensor]` is actually the return type would make a type-checker pass.

(Tuple is weird; `Sequence[tf.Tensor]` and `Iterable[tf.Tensor]` would have passed okay.)